ArchiverLib/Sources/ContentExtractorStore/ContentExtractorCache.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  ContentExtractorCache.swift
+//  ArchiverLib
+//
+//  Created by Claude on 31.10.25.
+//
+
+import CryptoKit
+import Foundation
+import OSLog
+import Shared
+
+/// Manages caching of content extraction results to improve performance
+@available(iOS 26, macOS 26, *)
+public actor ContentExtractorCache: Log {
+
+    // MARK: - Types
+
+    public struct CacheEntry: Codable, Sendable {
+        public let documentURL: URL
+        public let specification: String
+        public let tags: [String]
+        public let timestamp: Date
+
+        public init(documentURL: URL, specification: String, tags: [String], timestamp: Date = Date()) {
+            self.documentURL = documentURL
+            self.specification = specification
+            self.tags = tags
+            self.timestamp = timestamp
+        }
+    }
+
+    // MARK: - Properties
+
+    private let cacheDirectory: URL
+    private var isEnabled: Bool
+
+    // MARK: - Initialization
+
+    public init(cacheDirectory: URL? = nil, isEnabled: Bool = true) {
+        // Use app's cache directory by default (system can remove if needed)
+        if let cacheDirectory {
+            self.cacheDirectory = cacheDirectory
+        } else {
+            let defaultCache = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
```

Comment:
avoid force unwrapping by using this:

https://developer.apple.com/documentation/foundation/url/cachesdirectory